.github/workflows/build-test-ci.yml

```diff
@@ -45,7 +42,25 @@ jobs:
           --disable-maintainer-mode --disable-openmp
           ${{ matrix.features }}
       - name: make variant ${{ matrix.features }}
-        run: make CFLAGS="-O3 -Wall -Wshadow -pipe" V=1 check
+        run: >
+          export CFLAGS="-O3 -Wall -Wshadow -pipe"
+          export ASAN_OPTIONS="${ASAN_OPTIONS}:halt_on_error=1:abort_on_error=1:print_summary=1"
+          export UBSAN_OPTIONS="${UBSAN_OPTIONS}:halt_on_error=1:abort_on_error=1:print_summary=1:print_stacktrace=1"
+
+          case "$SANITIZER" in
+            none)
+               ;;
+            asan)
+               export CFLAGS="${CFLAGS} -ggdb3 -fsanitize=address"
```

Comment:
Is it useful to use -O3 here?  do we get useful stacks and analysis this way?  For valgrind we also run with `-g -pipe` to get understandable complaints